libfwupdplugin/fu-common.c

```diff
@@ -197,6 +198,88 @@ fu_common_check_full_disk_encryption(GError **error)
 	return TRUE;
 }
 
+/**
+ * fu_common_get_olson_timezone_id:
+ * @error: (nullable): optional return location for an error
+ *
+ * Gets the system Olson timezone ID, as used in the CLDR and ICU specifications.
+ *
+ * Returns: timezone string, e.g. `Europe/London` or %NULL on error
+ *
+ * Since: 1.9.7
+ **/
+gchar *
+fu_common_get_olson_timezone_id(GError **error)
+{
+	g_autofree gchar *sysconfdir = fu_path_from_kind(FU_PATH_KIND_SYSCONFDIR);
+	g_autofree gchar *fn_lz = fu_path_from_kind(FU_PATH_KIND_LOCALTIME);
+	g_autofree gchar *fn_tz = g_build_filename(sysconfdir, "timezone", NULL);
+	g_autoptr(GFile) file_lz = g_file_new_for_path(fn_lz);
+	g_autoptr(GFile) file_tz = g_file_new_for_path(fn_tz);
+
+	/* old Ubuntu, or no systemd */
+	g_debug("looking for timezone file %s", fn_tz);
+	if (g_file_query_exists(file_tz, NULL)) {
+		gsize bufsz = 0;
+		g_autofree gchar *buf = NULL;
+		g_auto(GStrv) lines = NULL;
+		if (!g_file_load_contents(file_tz, NULL, &buf, &bufsz, NULL, error))
+			return NULL;
+		lines = fu_strsplit(buf, bufsz, "\n", -1);
+		for (guint i = 0; lines[i] != NULL; i++) {
+			if (lines[i][0] == '\0')
+				continue;
+			if (g_str_has_prefix(lines[i], "#"))
+				continue;
+			return g_strdup(lines[i]);
+		}
+	}
+
+	/* Red Hat and *BSD use the last two sections of the symlink target */
```

Comment:
is that not what Ubuntu does too?
```
$ ls /etc/localtime -alh
lrwxrwxrwx 1 root root 35 Oct 12 16:23 /etc/localtime -> /usr/share/zoneinfo/America/Chicago
```

Reply:
IE I would think you can instead just look at both paths and do this instead of having a "read this file" or "look at symlink value"

Reply:
"new" Ubuntu does -- not sure how far that behaviour goes? Maybe the switch to systemd?

Reply:
How old do we really care about?  I see that in Ubuntu 22.04.

Reply:
Ohh, 20.04 seems "new enough" -- shall I rip out the super-legacy stuff now?

Reply:
Ripped out.